qcodes/instrument_drivers/tektronix/Keithley_2600_channels.py

```diff
@@ -146,18 +140,20 @@ def _time_trace(self, npts: int, dt: float, mode: str) -> np.ndarray:
             raise RuntimeError("No instrument attached to Parameter.")
 
         channel = self.instrument.channel
-        self.shape = (npts,)
-
-        script = ['{}.measure.count={}'.format(channel, self.shape[0]),
-        'oldint={}.measure.interval'.format(channel),
-        '{}.measure.interval={}'.format(channel, dt),
-        '{}.nvbuffer1.clear()'.format(channel),
-        '{}.measure.{}({}.nvbuffer1)'.format(channel, mode, channel),
-        '{}.measure.interval=oldint'.format(channel),
-        '{}.measure.count=1'.format(channel),
-        'format.data = format.REAL32',
-        'format.byteorder = format.LITTLEENDIAN',
-        'printbuffer(1, {}, {}.nvbuffer1.readings)'.format(self.shape[0], channel)]
+        npts = self.instrument.npts()
+        dt = self.instrument.dt()
+        mode = self.instrument.timetrace_mode()
+
+        script = ['{}.measure.count={}'.format(channel, npts),
+                  'oldint={}.measure.interval'.format(channel),
+                  '{}.measure.interval={}'.format(channel, dt),
+                  '{}.nvbuffer1.clear()'.format(channel),
+                  '{}.measure.{}({}.nvbuffer1)'.format(channel, mode, channel),
```

Comment:
Note that if we do go ahead and change the `mode` values to be `"current" ` and `"voltage"`, then this line needs to change.